### PR TITLE
Sync  ItCrossDomainTransaction tests from main

### PR DESCRIPTION
**Pull request by anpanigr**
Refer Jira OWLS-98531

Jenkin Run: 
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/10064 (individual)
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/10065 (all)

### CODE REVIEW

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItCrossDomainTransaction.java [hunk 1]
```diff
@@ -507,36 +520,27 @@ private static void createDomain(String domainUid, String domainNamespace, Strin
...
 
     // wait for the domain to exist
-    logger.info("Checking for domain custom resource in namespace {0}", domainNamespace);
-    testUntil(
-        domainExists(domainUid, DOMAIN_VERSION, domainNamespace),
-        logger,
-        "domain {0} to be created in namespace {1}",
-        domainUid,
-        domainNamespace);
+    logger.info("Check for domain custom resource in namespace {0}", domainNamespace);
```
Comment (sankarpn) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItCrossDomainTransaction.java:523]: Why are we rolling back these changes?

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItCrossDomainTransaction.java [hunk 2]
```diff
@@ -360,13 +373,14 @@ private static void buildApplicationsAndDomains() {
...
    */
+  @Order(1)
   @Test
   @DisplayName("Check cross domain transaction works")
   void testCrossDomainTransaction() {
 
     String curlRequest = String.format("curl -v --show-error --noproxy '*' "
-            + "http://%s/TxForward/TxForward?urls=t3://%s.%s:7001,t3://%s1.%s:8001,t3://%s1.%s:8001,t3://%s2.%s:8001",
-        hostAndPort, domain1AdminServerPodName, domain1Namespace,
+            + "http://%s:%s/TxForward/TxForward?urls=t3://%s.%s:7001,t3://%s1.%s:8001,t3://%s1.%s:8001,t3://%s2.%s:8001",
```
Comment (sankarpn) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItCrossDomainTransaction.java:382]: I believe this was added to support this test in OKD, why are we rolling it back?

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItCrossDomainTransaction.java [hunk 3]
```diff
@@ -383,8 +397,8 @@ void testCrossDomainTransaction() {
 
   /**
    * This test verifies a cross-domain transaction with re-connection.
-   * It makes sure the distributed transaction is completed successfully
-   * when a coordinator server is re-started after writing to transaction log
+   * It makes sure the disitibuted transaction is completed successfully
```
Comment (sankarpn) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItCrossDomainTransaction.java:400]: typo

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItCrossDomainTransaction.java [hunk 4]
```diff
@@ -383,8 +397,8 @@ void testCrossDomainTransaction() {
...
+   * when a coordinator server is re-started after writing to transcation log
```
Comment (sankarpn) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItCrossDomainTransaction.java:401]: typo

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItCrossDomainTransaction.java (hunk 2 above)
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItCrossDomainTransaction.java:382]: and similar changes below.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItCrossDomainTransaction.java [hunk 5]
```diff
@@ -550,30 +554,18 @@ private static void createDomain(String domainUid, String domainNamespace, Strin
       checkServiceExists(managedServerPrefix + i, domainNamespace);
     }
 
-    adminExtSvcRouteHost = createRouteForOKD(getExternalServicePodName(adminServerPodName), domainNamespace);
```
Comment (sankarpn) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItCrossDomainTransaction.java:553]: why are we removing the tests changes made for OKD?

**Review by sankarpn — CHANGES_REQUESTED** [review]

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItCrossDomainTransaction.java (hunk 1 above)
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItCrossDomainTransaction.java:522]: sync from the main branch

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItCrossDomainTransaction.java (hunk 2 above)
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItCrossDomainTransaction.java]: sync from the main branch

**Comment by maggiehe00** [comment]
Now the change makes sense. I will approve once the test is done.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItCrossDomainTransaction.java [hunk 6]
```diff
@@ -384,7 +406,7 @@ void testCrossDomainTransaction() {
   /**
    * This test verifies a cross-domain transaction with re-connection.
    * It makes sure the distributed transaction is completed successfully
-   * when a coordinator server is re-started after writing to transaction log
+   * when a coordinator server is re-started after writing to transation log
```
Comment (sankarpn) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItCrossDomainTransaction.java:409]: typo

**Review by sankarpn — CHANGES_REQUESTED** [review]
minor change

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItCrossDomainTransaction.java (hunk 5 above)
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItCrossDomainTransaction.java:553]: now synced form main branch

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItCrossDomainTransaction.java (hunk 6 above)
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItCrossDomainTransaction.java]: done

**Review by sankarpn — APPROVED** [review]
lgtm

**Review by maggiehe00 — APPROVED** [review]
LGTM

**Pull request merged**